src/datasets/tasks/text_classification.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass
+from typing import Dict, List
+
+from ..features import ClassLabel, Features, Value
+from .base import TaskTemplate
+
+
+@dataclass
+class TextClassification(TaskTemplate):
+    task = "text-classification"
+    input_schema = Features({"text": Value("string")})
+    label_schema = Features({"labels": ClassLabel})
+    labels: List[str]
+    text_column: str
+    label_column: str
+
+    def __init__(
+        self,
+        labels,
+        text_column="text",
+        label_column="labels",
+    ):
+        assert sorted(set(labels)) == sorted(labels), "Labels must be unique"
+        self.ordered_labels = sorted(labels)
+        self.label_schema["labels"] = ClassLabel(names=self.ordered_labels)
+        self.labels = labels
```

Comment:
Is this needed if we have `ordered_labels` ?

Reply:
good catch!

Reply:
actually i think we do need `labels` as an attribute of `TextClassification` because we rely on having access to it when we map between the class and dict (see the `from_dict` method).

Reply:
Maybe we should sort the labels in-place then?

I think two `TextClassification` tasks with the same set of labels should be perfectly equivalent. WDYT ?

Reply:
Maybe we should rename this method `__post_init__` in order not to interfere with the `dataclass` generated `__init__` method. See : https://docs.python.org/3/library/dataclasses.html#post-init-processing

Reply:
good idea! done :)